modules/javafx.graphics/src/main/java/javafx/scene/image/ImageView.java

```diff
@@ -826,13 +853,93 @@ public boolean isSettable(ImageView n) {
             }
         };
 
-         private static final List<CssMetaData<? extends Styleable, ?>> STYLEABLES;
-         static {
+        private static final CssMetaData<ImageView,Number> FIT_WIDTH =
+            new CssMetaData<ImageView,Number>("-fx-fit-width",
+                SizeConverter.getInstance(), 0.0) {
+
+                @Override
+                public boolean isSettable(ImageView n) {
+                    return n.fitWidth == null || !n.fitWidth.isBound();
+                }
+
+                @Override
+                public StyleableProperty<Number> getStyleableProperty(ImageView n) {
+                    return (StyleableProperty<Number>)n.fitWidthProperty();
+                }
+            };
+
+        private static final CssMetaData<ImageView,Number> FIT_HEIGHT =
+            new CssMetaData<ImageView,Number>("-fx-fit-height",
+                SizeConverter.getInstance(), 0.0) {
+
+                @Override
+                public boolean isSettable(ImageView n) {
+                    return n.fitHeight == null || !n.fitHeight.isBound();
+                }
+
+                @Override
+                public StyleableProperty<Number> getStyleableProperty(ImageView n) {
+                    return (StyleableProperty<Number>)n.fitHeightProperty();
+                }
+            };
+
+        private static final CssMetaData<ImageView,Boolean> SMOOTH =
+            new CssMetaData<ImageView,Boolean>("-fx-smooth",
+                new StyleConverter<String,Boolean>() {
+                    @Override
+                    public Boolean convert(ParsedValue<String, Boolean> value, Font font) {
+                        final String sval = value != null ? value.getValue() : null;
+                        return "true".equalsIgnoreCase(sval);
+                    }
+
+                },
+                SMOOTH_DEFAULT) {
+
+                @Override
+                public boolean isSettable(ImageView n) {
+                    return n.smooth == null || !n.smooth.isBound();
+                }
+
+                @Override
+                public StyleableProperty<Boolean> getStyleableProperty(ImageView n) {
+                    return (StyleableProperty<Boolean>)n.smoothProperty();
+                }
+            };
+
+        private static final CssMetaData<ImageView,Boolean> PRESERVE_RATIO =
+            new CssMetaData<ImageView,Boolean>("-fx-preserve-ratio",
+                new StyleConverter<String,Boolean>() {
+                    @Override
+                    public Boolean convert(ParsedValue<String, Boolean> value, Font font) {
+                        final String sval = value != null ? value.getValue() : null;
+                        return "true".equalsIgnoreCase(sval);
```

Comment:
I think we define "true" as a const

Reply:
Since 9 there is a public API javafx.css.converter.BooleanConverter that should be used.